platform-hub-api/app/models/ability.rb

```diff
@@ -5,5 +5,27 @@ def initialize(user)
 
     can :manage, :all if user.admin?
 
+
+    project_manager_checker = -> (target_user) do
+      # Find common projects between the two
+      target_user_projects = target_user.project_ids
+      user_projects = user.project_ids
+      common_projects = target_user_projects & user_projects
+
+      # Check to see if the user taking the action is a project manager of _any_
+      # common project they are both in
+      if common_projects
+        common_projects.any? do |p_id|
+          ProjectMembership.exists?(project_id: p_id, user_id: user.id, role: 'manager')
```

Comment:
Probably not without writing some ugly manual SQL :) I figured it's highly unlikely that two people will be in multiple projects, so the probability of this query being made more than once is very low :)